File: src/messaging-system/execution/ExecutionStatus.ts
/**
 * ExecutionStatus — C.4
 * Esito complessivo dell'esecuzione. SUCCESS | FAILED | SKIPPED.
 */

export const EXECUTION_STATUS = ['SUCCESS', 'FAILED', 'SKIPPED'] as const;

export type ExecutionStatus = (typeof EXECUTION_STATUS)[number];

File: flutter_app/lib/media_execution/execution_status.dart
// Media Execution — Execution status. Pure enum; no runtime logic.

/// Status of a physical operation execution.
/// Pure enum; no internal logic; no runtime reference.
enum ExecutionStatus {
  /// Operation completed successfully.
  success,

  /// Operation failed.
  failure,

  /// Operation was skipped (e.g., already completed or not applicable).
  skipped,
}
